logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/GetConversationVerificationStatusUseCase.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.conversation
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.conversation.Conversation
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.conversation.MLSConversationRepository
+import com.wire.kalium.logic.data.conversation.ConversationVerificationStatus
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.featureFlags.FeatureSupport
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.flatMap
+import com.wire.kalium.logic.functional.getOrElse
+import com.wire.kalium.logic.functional.map
+import com.wire.kalium.logic.kaliumLogger
+import com.wire.kalium.util.KaliumDispatcher
+import com.wire.kalium.util.KaliumDispatcherImpl
+import kotlinx.coroutines.withContext
+
+/**
+ * Operation that fetches [ConversationVerificationStatus] of the specific conversation
+ *
+ * @param conversationId
+ * @return [ConversationVerificationStatusResult.Failure] when error occurred.
+ * [ConversationVerificationStatusResult.Success] with [ConversationVerificationStatus] and [ConversationProtocol] of the conversation
+ * in other cases.
+ */
+interface GetConversationVerificationStatusUseCase {
+    suspend operator fun invoke(conversationId: ConversationId): ConversationVerificationStatusResult
+}
+
+class GetConversationVerificationStatusUseCaseImpl(
+    private val featureSupport: FeatureSupport,
+    private val clientRepository: ClientRepository,
+    private val conversationRepository: ConversationRepository,
+    private val mlsConversationRepository: MLSConversationRepository,
+    kaliumDispatcher: KaliumDispatcher = KaliumDispatcherImpl
+) : GetConversationVerificationStatusUseCase {
+    private val dispatcher = kaliumDispatcher.io
+
+    override suspend fun invoke(conversationId: ConversationId): ConversationVerificationStatusResult = withContext(dispatcher) {
+        if (!featureSupport.isMLSSupported || !clientRepository.hasRegisteredMLSClient().getOrElse(false)) {
```

Comment:
but there is a case, mls is disabled, but we have an MLS conversation from when the MLS was enabled [ mls can be enabled and disabled]
and if mls is disabled we should still get the non-mls verification status: